addOns/retest/src/main/java/org/zaproxy/addon/retest/ExtensionRetest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2014 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.retest;
+
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.zaproxy.addon.automation.AutomationPlan;
+
+public class ExtensionRetest extends ExtensionAdaptor {
+
+    public static final String NAME = "ExtensionRetest";
+
+    private RetestMenu retestMenu;
+    private RetestAPI retestApi;
+    private RetestDialog retestDialog;
+
+    private static final Logger LOGGER = LogManager.getLogger(ExtensionRetest.class);
+
+    public ExtensionRetest() {
+        super(NAME);
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        retestApi = new RetestAPI(this);
+        extensionHook.addApiImplementor(retestApi);
+
+        if (getView() != null) {
+            extensionHook.getHookMenu().addPopupMenuItem(getRetestMenu());
+        }
+    }
+
+    @Override
+    public boolean canUnload() {
+        return true;
+    }
+
+    @Override
+    public void unload() {
+        super.unload();
+        if (retestDialog != null) {
+            retestDialog.dispose();
+        }
+        retestMenu.removeAll();
```

Comment:
Remove the retestMenu itself?

Reply:
The menu will be removed automatically by core (was added with the hook). Is this really needed? (In any case it would need a view check.)